apps/gateway/src/routes/api/v1/projects/:projectId/commits/:commitUuid/documents/handlers/run.test.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  apiKeys,
+  ChainEventTypes,
+  database,
+  factories,
+  LATITUDE_EVENT,
+  mergeCommit,
+  Result,
+  streamText,
+} from '@latitude-data/core'
+import app from '$/index'
+import { eq } from 'drizzle-orm'
+import { describe, expect, it, vi, type Mock } from 'vitest'
+
+vi.mock('@latitude-data/core', async (importOriginal) => {
+  const original = (await importOriginal()) as typeof importOriginal
+
+  return {
+    ...original,
+    streamText: vi.fn(),
+  }
+})
+
+describe('POST /run', () => {
+  describe('unauthorized', () => {
+    it('fails', async () => {
+      const res = await app.request(
+        '/api/v1/projects/1/commits/asldkfjhsadl/documents/run',
+        {
+          method: 'POST',
+          body: JSON.stringify({
+            documentPath: '/path/to/document',
+          }),
+        },
+      )
+
+      expect(res.status).toBe(401)
+    })
+  })
+
+  describe('authorized', () => {
+    it('succeeds', async () => {
+      const stream = new ReadableStream({
+        start(controller) {
+          controller.enqueue({
+            event: LATITUDE_EVENT,
+            data: {
+              type: ChainEventTypes.Complete,
+              response: {
+                text: 'Hello',
+                usage: {},
+              },
+            },
+          })
+
+          controller.close()
+        },
+      })
+
+      const response = new Promise((resolve) => {
+        resolve({ text: 'Hello', usage: {} })
+      })
+
+      ;(streamText as Mock).mockReturnValue(
```

Comment:
Do this way please. Is less dirty
https://github.com/latitude-dev/latitude-llm/blob/main/apps/web/src/actions/commits/fetchCommitsByProjectAction.test.ts#L8-L15